shuffle/cli/src/transactions.rs

```diff
@@ -0,0 +1,227 @@
+    let seq_num = get_sequence_number(&client, &network, address).await?;
+    let start = max(seq_num as i64 - 10, 0);
+
+    let resp = client
+        .get(path.as_str())
+        .query(&[("start", start.to_string().as_str())])
+        .query(&[("limit", "10")])
+        .send()
+        .await?;
+
+    let json: serde_json::Value = serde_json::from_str(resp.text().await?.as_str())?;
+
+    let all_transactions = json
+        .as_array()
+        .ok_or_else(|| anyhow::anyhow!("Failed to get transactions"))?;
+
+    for txn_index in 0..all_transactions.len() {
+        write_into(&mut io::stdout(), &json[txn_index])?;
+        println!();
+        println!();
+    }
+
+    let mut old_seq_num = 0;
+    if !all_transactions.is_empty() {
+        old_seq_num = all_transactions
+            .get(all_transactions.len() - 1)
+            .ok_or_else(|| anyhow!("Couldn't get last transaction"))?["sequence_number"]
+            .to_string()
+            .replace('"', "")
+            .parse::<u64>()?;
+    }
+    if tailing {
+        // listening for incoming transactions
+        loop {
+            thread::sleep(time::Duration::from_millis(1000));
+            let resp = get_account_txns_response(&client, &path, old_seq_num, 100).await?;
+            let json_with_txns: serde_json::Value =
+                serde_json::from_str(resp.text().await?.as_str())?;
+            let size_of_json = json_with_txns
+                .as_array()
+                .ok_or_else(|| anyhow!("Couldn't convert to array"))?
+                .len();
+
+            // checking if there are transactions
+            if size_of_json > 0 {
+                let last_txn_seq_num = &json_with_txns
+                    .get(size_of_json - 1)
+                    .ok_or_else(|| anyhow!("Couldn't get last transaction"))?["sequence_number"]
+                    .to_string()
+                    .replace('"', "")
+                    .parse::<u64>()?;
+                old_seq_num = write_out_txns(
+                    json_with_txns,
+                    *last_txn_seq_num,
+                    old_seq_num,
+                    &mut io::stdout(),
+                )?;
+            }
+        }
+    }
+    Ok(())
+}
+
+pub fn write_out_txns<W: Write>(
+    json: Value,
+    last_txn_seq_num: u64,
+    old_seq_num: u64,
+    mut stdout: W,
+) -> Result<u64> {
+    // Only want to write the first transaction (seq num: 0) or
+    // transactions where the sequence number changes
+    if last_txn_seq_num == 0 || last_txn_seq_num > old_seq_num {
+        let all_transactions = json
+            .as_array()
+            .ok_or_else(|| anyhow::anyhow!("Failed to get transactions"))?;
+
+        // setting to 1 to exclude the index of the txn we've already printed out
+        let mut start_ind = 1;
+        if last_txn_seq_num == 0 {
+            // setting to 0 to include the first txn.
+            // Only needed for the very first txn
```

Comment:
Again, why? Is it a 1 vs 0 based indexing issue?

Reply:
Changed so i'm not indexing. handling this edge case outside of this method.